components/templates/src/global_fns/load_data.rs

```diff
@@ -0,0 +1,403 @@
+extern crate toml;
+extern crate serde_json;
+
+use utils::fs::{read_file, is_path_in_directory, get_file_time};
+
+use std::hash::{Hasher, Hash};
+use std::fmt;
+use std::collections::hash_map::DefaultHasher;
+use reqwest::{Client, header};
+use url::Url;
+
+use std::path::PathBuf;
+use std::sync::{Arc, Mutex};
+
+
+use csv::Reader;
+use std::collections::HashMap;
+use tera::{GlobalFn, Value, from_value, to_value, Result, Map};
+
+static GET_DATA_ARGUMENT_ERROR_MESSAGE: &str = "`load_data`: requires EITHER a `path` or `url` argument";
+
+enum DataSource {
+    Url(Url),
+    Path(PathBuf)
+}
+
+#[derive(Debug)]
+enum OutputFormat {
+    TOML,
+    JSON,
+    CSV,
```

Comment:
I wouldn't uppercase them even if they are acronyms to be consistent with Rust style (https://doc.rust-lang.org/1.0.0/style/style/naming/README.html)